keanu-project/src/main/java/io/improbable/keanu/util/Py4jUtils.java

```diff
@@ -0,0 +1,43 @@
+package io.improbable.keanu.util;
+
+import lombok.experimental.UtilityClass;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+@UtilityClass
+public class Py4jUtils {
+
+    public byte[] toByteArray(double[] doubleArray) {
+        ByteBuffer doubleBuffer = ByteBuffer.allocate(Double.SIZE / Byte.SIZE * doubleArray.length);
+        // Java defaults to BIG_ENDIAN. LITTLE_ENDIAN is what is expected by numpy to construct an ndarray.
+        doubleBuffer.order(ByteOrder.LITTLE_ENDIAN);
+        for (double element : doubleArray) {
+            doubleBuffer.putDouble(element);
+        }
+        byte[] byteArray = doubleBuffer.array();
+        return byteArray;
+    }
+
+    public byte[] toByteArray(int[] integerArray) {
+        ByteBuffer integerBuffer = ByteBuffer.allocate(Integer.SIZE / Byte.SIZE * integerArray.length);
+        // Java defaults to BIG_ENDIAN. LITTLE_ENDIAN is what is expected by numpy to construct an ndarray.
+        integerBuffer.order(ByteOrder.LITTLE_ENDIAN);
+        for (int element : integerArray) {
+            integerBuffer.putInt(element);
+        }
+        byte[] byteArray = integerBuffer.array();
+        return byteArray;
+    }
+
+    public byte[] toByteArray(boolean[] booleanArray) {
+        ByteBuffer booleanBuffer = ByteBuffer.allocate(booleanArray.length);
+        booleanBuffer.order(ByteOrder.LITTLE_ENDIAN);
+        for (boolean element : booleanArray) {
+            byte booleanByte = element ? (byte) 0x01 : 0x00;
```

Comment:
does this do the cast on the return of the conditional? that's cool. I thought you'd have to do `(byte) 0x01 : (byte) 0x00`

Reply:
So did I, and then did the first one and it stopped complaining